### DISCUSSION TITLE
HTTP 401 from API requests when going through Nginx with basic authentication

### DISCUSSION THREAD

**Discussion by EvanED** [requarks/wiki#4226] [Help / Questions]
I am shepherding wiki.js via nginx using the following virtual configuration file (with a little anonymization):

```
server {
    server_name wiki.example.com wiki.example.name;

    auth_basic "My Wiki";
    auth_basic_user_file "/etc/nginx/sites-available/wiki.htpasswd";

    client_max_body_size 300M;

    location / {
        # Set proxy headers
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";

        proxy_pass http://localhost:3000;
    }

    listen [::]:443 ssl;
    listen 443 ssl; # managed by Certbot
    ssl_certificate <..>.;
    ssl_certificate_key <...>;
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
}
```

This conf file is basically taken from another server I do the same thing with, so I'm not sure I've got all the `proxy_*` fields correct and I'm not sure all are necessary here.

Anyway, the site *kind of* loads correctly, however I get a "Network Error: JSON.parse: unexpected character at line 1 column 1 of the JSON data" error banner. The cause of that is because the POST request to `https://wiki.example.com/graphql`  responds with a 401
Unauthorized.

I can't tell how much this is a wiki.js question and how much this is an nginx question, but can anyone help me out? Sysadmin-y stuff isn't my forte.

**Comment by KBCACheia** [comment]
I am getting the same error without authentication.
